dbt_semantic_interfaces/parsing/schemas.py

```diff
@@ -242,6 +242,34 @@
 }
 
 
+time_spine_table_configuration_schema = {
+    "$id": "time_spine_table_configuration",
+    "type": "object",
+    "properties": {
+        "location": {"type": "string"},
+        "column_name": {"type": "string"},
+        "grain": {"enum": time_granularity_values},
+    },
+    "additionalProperties": False,
+    "required": ["location", "column_name", "grain"],
+}
+
+
+project_configuration_schema = {
+    "$id": "project_configuration",
+    "type": "object",
+    "properties": {
+        "dsi_package_version": {"type": "string"},
```

Comment:
Can you clarify how that would work? e.g. we release DSI v0.2, but the customer wrote and committed configs based on DSI v0.1. If the server is upgraded and deserializes the files, wouldn't the version be incorrect?

Reply:
Deserialization and parsing logic is not the same. The JSON schema is only used for validating the yaml prior to parsing it to the object. Specifically it is invoked in the from of `<object_validator>.validate(...)` ([example](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/739fae883f991b92ad6db877223e5068a241947d/dbt_semantic_interfaces/parsing/dir_to_model.py#L297)). Deserialization is the act of calling `<object_type>.parse_raw(...yaml...)` deserialization by itself doesn't call json schema validation. Additionally in our json schema we explicitly disallow extra keys thus if someone tries to set a `dsi_package_version` we want things to fail. Because deserialization doesn't do json schema validation, this is not a problem. Thus by using using a pydantic `validate` setter we can handle both situtiations. When going through parsing, the `dsi_package_version` should be `None`, so we auto generate it. When deserializing, the value for `dsi_package_version` would be included in the serialization and thus will be used.